src/components/checkbox/examples/checkbox.tsx

```diff
@@ -15,6 +15,11 @@ export class CheckboxExample {
     @State()
     private value: boolean = false;
 
+    @State()
+    private indeterminate: boolean = false;
```

Comment:
Hmm… 🤔

If I set this to `true`, I can't programmatically set the checkbox to _not_ indeterminate again 😮

Going for lunch now, will look into it again when I'm back if you haven't already found something out by then 😄

Reply:
That was a bug I had for two days and then I haven't been able to reproduce it after restarting the stencil dev server. Please please show me :)

Reply:
Ok, so my idea worked. The MDCCheckbox sets the property (not attribute) indeterminate=true on the native input and that will keep the indeterminate style because it still matches the selector "nativeinput:indeterminate". Only updating the CSS classes mdc-checkbox--indeterminate was not enough when using indeterminate={true} on the initial render. 

The fix I applied isn't pretty but it works. Instead of manually resetting the indeterminate state of the native element I added a watcher that unsets indeterminate on the MDCCheckbox instance instead (letting it do the job of toggling the state, which also unsets things like `aria-checked="mixed"`).